index.js

```diff
@@ -1,139 +1,158 @@
-        if (parsed && parsed.next && parsed.next.url) {
-          fetchPageOfDataAndFilter(parsed.next.url).then(newEvents => {
-            return resolve(importantEvents.concat(newEvents));
+      .then((response, err) => {
+        const parsed = parse(response.headers.get("link"));
+        let importantEvents = [];
+        response
+          .json()
+          .then((json, err) => {
+            const filteredForImportant = filterResponseForImportantEvents(json);
+            importantEvents = importantEvents.concat(filteredForImportant);
+
+            if (!(parsed && parsed.next && parsed.next.url)) {
+              return resolve(importantEvents);
+            }
+
+            fetchPageOfDataAndFilter(parsed.next.url).then(newEvents =>
+              resolve(importantEvents.concat(newEvents))
+            );
+          })
+          .catch(err => {
+            console.log("Error turning response into JSON:", err);
           });
-        } else {
-          return resolve(importantEvents);
-        }
       })
-      .catch(err => {console.log("Error turning response into JSON:", err);})
-    })
-    .catch(err => console.log("ERROR GRABBING INFO FROM GITHUB!", err))
+      .catch(err => console.log("ERROR GRABBING INFO FROM GITHUB!", err));
   });
 }
 
+const importantEventTypes = [
+  "CommitCommentEvent",
+  "IssueCommentEvent",
+  "IssuesEvent",
+  "PullRequestEvent",
+  "PullRequestReviewEvent",
+  "PullRequestReviewCommentEvent"
+];
+
 function filterResponseForImportantEvents(allEventsFromFetch) {
-  let arrayOfImportantEvents = [];
-  for(let i=0; i<allEventsFromFetch.length; i++) {
-    let event = allEventsFromFetch[i];
-    switch(event.type) {
-      case 'CommitCommentEvent':
-      case 'IssueCommentEvent':
-      case 'IssuesEvent':
-      case 'PullRequestEvent':
-      case 'PullRequestReviewEvent':
-      case 'PullRequestReviewCommentEvent':
-        arrayOfImportantEvents.push(event);
-        break;
+  const arrayOfImportantEvents = [];
+  for (let i = 0; i < allEventsFromFetch.length; i++) {
+    const event = allEventsFromFetch[i];
+    if (importantEventTypes.includes(event.type)) {
+      arrayOfImportantEvents.push(event);
     }
   }
-  return arrayOfImportantEvents
+  return arrayOfImportantEvents;
 }
 
 function createIdObjects(row, idObject, importantEvents) {
   idObject.alternateId = row[alternateIdColumnNumber];
   idObject.github = row[githubIdColumnNumber];
   idObject.contributions = importantEvents;
-  arrayOfIdObjects.push(idObject)
+  arrayOfIdObjects.push(idObject);
 }
 
 function filterContributorByTime(idObject, dates) {
-  const timeWindow = moment.range([moment.utc(`${dates[0]}`, 'YYYY-MM-DD hh:mm'), moment.utc(`${dates[1]}`, 'YYYY-MM-DD hh:mm')]);
+  const timeWindow = moment.range([
+    moment.utc(`${dates[0]}`, "YYYY-MM-DD hh:mm"),
+    moment.utc(`${dates[1]}`, "YYYY-MM-DD hh:mm")
+  ]);
   const contribsByUser = [];
```

Comment:
I didn't remove this `const`, just in case, but: what it's really doing? Is not being used in any place. Shouldn't it also be removed?